src/hitch.c

```diff
@@ -581,7 +581,11 @@ create_shcupd_socket()
 					fail("{ioctl: SIOCGIFINDEX}");
 				}
 
+#ifdef __FreeBSD__
```

Comment:
Wouldn't using the posix function `if_nametoindex` be better?  By the way, the corresponding code further down for IPv6 should also be changed.